### DISCUSSION TITLE
First build always configures cmake and changes build type to Debug

### DISCUSSION THREAD

**Discussion by boxerab** [microsoft/vscode-cmake-tools#4184] [General]
Here are my settings

```
{
    "git.openRepositoryInParentFolders": "never",
    "security.workspace.trust.untrustedFiles": "open",
    "cmake.pinnedCommands": [
        "workbench.action.tasks.configureTaskRunner",
        "workbench.action.tasks.runTask"
    ],
    "cmake.showOptionsMovedNotification": false,
    "cmake.configureOnOpen": false,
    "cmake.clearOutputBeforeBuild": false,
    "cmake.cacheInit": null,
    "cmake.buildBeforeRun": false,
    "cmake.configureSettings": {
    
        
    }
}
```

I don't want cmake tools to change my configuration at all unless I explicitly change it.
However, the first build after startup always changes build type back to Debug. 

How can I disable this?  Or is it a bug.

Thanks!

**Comment by gcampbell-msft** [comment]
@boxerab Could you provide more information about your repro? A full repro project along with steps would be most beneficial. Are you selecting a configure preset?

**Comment by boxerab** [comment]
@gcampbell-msft thanks. I am using vscodium, btw, in case that makes a difference, but it shouldn't
Here is a more pinpointed description:
1. open vscodium
2. edit cmake cache ui
3. change build type to release
4. save
5. exit
6. reopen 
7. edit cmake cache ui
8. build type has now been set back to debug

**Reply by gcampbell-msft** [reply]
Oh interesting, based on your steps, it sounds like this is actually #2795

**Comment by boxerab** [comment]
It does sound similar. In my case, I enter Release so the string isn't empty.

**Reply by gcampbell-msft** [reply]
Could you please follow up on that bug and copy your repro steps there? I'll close this discussion and track the issue there. Thanks!
